scripts/prepare_release.py

```diff
@@ -0,0 +1,803 @@
+                      if 'Security' in changelog
+                      else 'None.')
+        with open(self.checksum_path, encoding='ascii') as inp:
+            checksums = inp.read()
+        # We currently emit a single archive file. If we switch to having
+        # multiple files, tweak this definition and the grammar where
+        # it's used.
+        m = re.search(r'\S+$', checksums)
+        if m is None:
+            raise Exception('Unable to determine archive file name')
+        archive_name = m.group(0)
+        # The very long lines here are deliberate. GitHub treats newlines
+        # in markdown as line breaks, not as spaces, so an ordinary paragraph
+        # needs to be in a single Python logical line.
+        return f'''\
+## Description
+
+This release of {self.info.product_human_name} provides new features, bug fixes and minor enhancements. \
+{'This release includes fixes for security issues.' if 'Security' in changelog else ''}
+
+## Security Advisories
+
+{advisories}
+
+## Release Notes
+
+{changelog[None]}
+
+## Who should update
+
+We recommend all users should update to take advantage of the bug fixes contained in this release at an appropriate point in their development lifecycle.
+
+## Note
+
+:grey_exclamation: `{archive_name}` is our official release file. `source.tar.gz` and `source.zip` are automatically generated snapshots that GitHub is generating. They do not include submodules or generated files, and [cannot be configured](https://github.com/orgs/community/discussions/6003).
+
+## Checksum
+
+The SHA256 hashes for the archives are:
+```
+{checksums}
+```
+'''
+
+    def run(self) -> None:
+        content = self.notes()
+        with open(self.artifact_path('.md'), 'w') as out:
+            out.write(content)
+
+
+ALL_STEPS = [
+    AssembleChangelogStep,
+    BumpVersionStep,
+    ArchiveStep,
+    NotesStep,
+] #type: Sequence[typing.Type[Step]]
+
+
+def run(options: Options,
+        top_dir: str,
+        from_: Optional[str] = None,
+        to: Optional[str] = None) -> None:
+    """Run the release process (or a segment thereof)."""
+    info = Info(top_dir, options)
+    from_reached = (from_ is None)
+    for step_class in ALL_STEPS:
+        step = step_class(options, info)
+        if not from_reached:
+            if step.name() != from_:
+                continue
+            from_reached = True
+        step.assert_preconditions()
+        step.run()
+        if step.name() == to:
+            break
+
+def main() -> None:
+    """Command line entry point."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    # Options that affect information gathering, or that affect the
+    # behavior of one or more steps, should have an associated field
+    # in the Options class.
+    parser.add_argument('--directory',
+                        default=os.curdir,
+                        help='Product toplevel directory')
+    parser.add_argument('--artifact-directory', '-a',
```

Comment:
Similarly we are beiong inconsistent with the short arguments. We should have short versions for most arguments.